src/cursor/cur_std.c

```diff
@@ -1176,42 +1193,52 @@ __wt_cursor_largest_key(WT_CURSOR *cursor)
 int
 __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CONFIG_ITEM cval;
     WT_CURSOR_BTREE *cbt;
     WT_DECL_RET;
     WT_ITEM key;
     WT_SESSION_IMPL *session;
     int exact;
-    bool inclusive;
+    char *cfg;
+    bool inclusive, have_action;
 
+    cfg = NULL;
     cbt = (WT_CURSOR_BTREE *)cursor;
     exact = 0;
-    inclusive = false;
+    inclusive = true;
+    have_action = false;
 
-    CURSOR_API_CALL_CONF(cursor, session, bound, config, cfg, NULL);
+    CURSOR_API_CALL(cursor, session, bound, NULL);
 
     if (CUR2BT(cursor)->type == BTREE_COL_FIX)
         WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with fixed column store.");
 
     if (F_ISSET(cursor, WT_CURSTD_PREFIX_SEARCH))
         WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with prefix search.");
 
-    WT_ERR(__wt_config_gets(session, cfg, "action", &cval));
-    if (WT_STRING_MATCH("set", cval.str, cval.len)) {
-        WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
-        inclusive = cval.val != 0;
+    if (config == NULL || config[0] == '\0')
+        WT_ERR_MSG(session, EINVAL, "an empty config is not valid when setting or clearing bounds");
 
-        /* Check that bound is set with action set configuration. */
-        WT_ERR(__wt_config_gets(session, cfg, "bound", &cval));
-        if (cval.len == 0)
-            WT_ERR_MSG(session, EINVAL, "setting bounds must require the bound configuration set");
+    /* Action is default to "set". */
+    if (CONFIG_VALUE_FOR(config, "action", cfg) != NULL)
+        have_action = true;
 
+    if (!have_action || WT_PREFIX_MATCH(cfg, "set")) {
         if (WT_CURSOR_IS_POSITIONED(cbt))
             WT_ERR_MSG(session, EINVAL, "setting bounds on a positioned cursor is not allowed");
 
         /* The cursor must have a key set to place the lower or upper bound. */
         WT_ERR(__cursor_checkkey(cursor));
-        if (WT_STRING_MATCH("upper", cval.str, cval.len)) {
+
+        if (CONFIG_VALUE_FOR(config, "inclusive", cfg) != NULL) {
```

Comment:
np: you could combine the both if statements into one here

Reply:
e0340ab10